org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/GradleProjectImporter.java

```diff
@@ -107,13 +113,37 @@ private void importDir(Path rootFolder, IProgressMonitor monitor) {
 	}
 
 	public static GradleDistribution getGradleDistribution(Path rootFolder) {
-		if (JavaLanguageServerPlugin.getPreferencesManager() != null && JavaLanguageServerPlugin.getPreferencesManager().getPreferences().isGradleWrapperEnabled() && Files.exists(rootFolder.resolve("gradlew"))) {
-			return GradleDistribution.fromBuild();
+		PreferenceManager preferencesManager = JavaLanguageServerPlugin.getPreferencesManager();
+		if (preferencesManager != null && preferencesManager.getPreferences().isGradleWrapperEnabled() && Files.exists(rootFolder.resolve("gradlew"))) {
+			WrapperValidator validator = new WrapperValidator();
+			try {
+				ValidationResult result = validator.checkWrapper(rootFolder.toFile().getAbsolutePath());
+				if (result.isValid()) {
+					WrapperGradleDistribution gradleDistribution = GradleDistribution.fromBuild();
+					return gradleDistribution;
+				} else {
+					if (!WrapperValidator.getDisallowed().contains(result.getSha256())) {
+						ProjectsManager pm = JavaLanguageServerPlugin.getProjectsManager();
+						if (pm != null && pm.getConnection() != null) {
+							//@formatter:off
+							String message = "__GRADLE_WRAPPER_SHA256__";
+							ActionableNotification notification = new ActionableNotification()
```

Comment:
you need to check if client supports actionable notifications and specifically supports gradle wrapper management

Reply:
else need to send a regular notification to the client

Reply:
> ... specifically supports gradle wrapper management

We check it at https://github.com/eclipse/eclipse.jdt.ls/pull/1434/files/60b458349b780484594682b93dffaf2db16e2f0f#diff-e696f471c869cd239fd5dbb142db81efR117

Reply:
Fixed.